### PR TITLE
Leap/ui improvement

### PR DESCRIPTION
**Pull request by simleap**
- change callout communes from selectize to normal checkboxes
- allow user to create callout with restrict communes
- reverse english khmer for locations

pivotal: https://www.pivotaltracker.com/story/show/158136815
screenshot: https://goo.gl/htUKEx

### CODE REVIEW

File: app/assets/javascripts/dashboard/callout.js [hunk 1]
```diff
@@ -1,9 +1,122 @@
...
+      }
+    });
+  };
+
+  onCommuneCheckboxChange = function () {
+    _this.$districtsCommunes.on('change', '.commune-ids', function () {
+      var $district = findDistrictOf($(this).val());
+
+      if ($(this).is(':checked')) {
+        $district.prop('checked', true);
```
Comment (samnang) [inline app/assets/javascripts/dashboard/callout.js]: make_emergency_call

Should we use this https://developer.mozilla.org/en-US/docs/Web/CSS/:indeterminate instead?

File: app/models/callout.rb [hunk 2]
```diff
@@ -39,6 +40,8 @@ class Callout < ApplicationRecord
 
   validates :call_flow_logic,
             presence: true
+  validates :commune_ids,
+            array_inclusion: { in: :permitted_commune_ids }, if: :created_by
```
Comment (samnang) [inline app/models/callout.rb]:
```ruby

validate :province_permitted

private

def province_permitted
  return if created_by.blank?
  return if created_by.province_ids.include?(province_id)
  errors.add(:commune_ids, :inclusion)
end
```

File: app/models/callout.rb [hunk 3]
```diff
@@ -76,4 +79,14 @@ class Callout < ApplicationRecord
       )
     end
   end
+
+  private
+
+  def permitted_commune_ids
```
Comment (samnang) [inline app/models/callout.rb]: remove this

File: app/validators/array_inclusion_validator.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class ArrayInclusionValidator < ActiveModel::EachValidator
```
Comment (samnang) [inline app/validators/array_inclusion_validator.rb]: Delete this

File: app/assets/javascripts/dashboard/callout.js [hunk 5]
```diff
@@ -1,9 +1,122 @@
...
+    return '<div class="form-check">' +
+    '<input class="form-check-input ' + type + '-ids" type="checkbox" ' +
+    'value=' + object.id + ' name="callout[' + type + '_ids][]" ' +
+    'id="callout_' + type + '_' + object.id + '">' +
+    '<label class="string optional form-check-label"' +
+    'for="callout_' + type + '_' + object.id + '">'
+    + object.name_km + '&nbsp;' + object.name_en + '</label></div>';
+  };
+
+  onCheckAll = function () {
```
[the remark this answers is not in the document]
Comment (simleap) [reply app/assets/javascripts/dashboard/callout.js]: @samnang, do you mean we don't need check all checkbox?

[the remark this answers is not in the document]
Comment (samnang) [reply app/assets/javascripts/dashboard/callout.js]: yes, we don't need that `Check All` checkbox.

**Comment by simleap** [comment]
@samnang, i have updated this PR.

**Pull request merged**